src/rgw/rgw_user.cc

```diff
@@ -2602,7 +2602,7 @@ class RGWUserMetadataHandler : public RGWMetadataHandler {
     for (iter = unfiltered_keys.begin(); iter != unfiltered_keys.end(); ++iter) {
       string& k = *iter;
 
-      if (k.find(".buckets") == string::npos) {
+      if (k.find(".buckets") != string::npos) {
```

Comment:
the `RGWUserMetadataHandler` is used for several things, including the `radosgw-admin user list` command - this change will cause that command to return the wrong results

i also don't quite understand how this listing would be causing `cls_user_get_header()` to fail